src/ui/bagl_display.c

```diff
@@ -187,49 +187,110 @@ UX_STEP_NOCB(ux_display_amount_step,
-
-UX_FLOW(ux_display_cancel_transaction_flow,
-        &ux_display_review_step,
-        &ux_display_type_step,
+        &ux_display_to_step,
         &ux_display_nonce_step,
         &ux_display_gas_price_step,
         &ux_display_gas_limit_step,
         &ux_display_approve_step,
         &ux_display_reject_step);
 
+static void handle_display_legacy() {
+    ux_flow_init(0, ux_display_legacy_transaction_flow, NULL);
+}
+
+static void handle_display_value_transfer() {
+    ux_flow_step_t static const *ux_display_flow[MAX_FLOW_STEPS];
+    int current_step = 0;
+
+    ux_display_flow[current_step++] = &ux_display_review_step;
+    ux_display_flow[current_step++] = &ux_display_type_step;
+    ux_display_flow[current_step++] = &ux_display_amount_step;
+    ux_display_flow[current_step++] = &ux_display_to_step;
+    ux_display_flow[current_step++] = &ux_display_nonce_step;
+    ux_display_flow[current_step++] = &ux_display_gas_price_step;
+    ux_display_flow[current_step++] = &ux_display_gas_limit_step;
+
+    if (G_context.tx_info.transaction.ratio != 0) {
+        ux_display_flow[current_step++] = &ux_display_fee_ratio_step;
+    }
+
+    ux_display_flow[current_step++] = &ux_display_approve_step;
+    ux_display_flow[current_step++] = &ux_display_reject_step;
+    ux_display_flow[current_step++] = FLOW_END_STEP;
+
+    ux_flow_init(0, ux_display_flow, NULL);
+}
+
+static void handle_display_smart_contract_deploy_display() {
+    ux_flow_step_t static const *ux_display_flow[MAX_FLOW_STEPS];
+    int current_step = 0;
+
+    ux_display_flow[current_step++] = &ux_display_review_step;
+    ux_display_flow[current_step++] = &ux_display_type_step;
+    ux_display_flow[current_step++] = &ux_display_amount_step;
+    ux_display_flow[current_step++] = &ux_display_nonce_step;
+    ux_display_flow[current_step++] = &ux_display_gas_price_step;
+    ux_display_flow[current_step++] = &ux_display_gas_limit_step;
+
+    if (G_context.tx_info.transaction.ratio != 0) {
+        ux_display_flow[current_step++] = &ux_display_fee_ratio_step;
+    }
+
+    ux_display_flow[current_step++] = &ux_display_approve_step;
+    ux_display_flow[current_step++] = &ux_display_reject_step;
+    ux_display_flow[current_step++] = FLOW_END_STEP;
+
+    ux_flow_init(0, ux_display_flow, NULL);
+}
+
+static void handle_display_smart_contract_execution_display() {
+    ux_flow_step_t static const *ux_display_flow[MAX_FLOW_STEPS];
+    int current_step = 0;
+
+    ux_display_flow[current_step++] = &ux_display_review_step;
+    ux_display_flow[current_step++] = &ux_display_type_step;
+    ux_display_flow[current_step++] = &ux_display_amount_step;
+    ux_display_flow[current_step++] = &ux_display_smart_contract_step;
+    ux_display_flow[current_step++] = &ux_display_nonce_step;
+    ux_display_flow[current_step++] = &ux_display_gas_price_step;
+    ux_display_flow[current_step++] = &ux_display_gas_limit_step;
+
+    if (G_context.tx_info.transaction.ratio != 0) {
```

Comment:
it is odd that the screen is not displayed since there is this condition here. to investigate